src/ripple/app/paths/impl/StrandFlow.h

```diff
@@ -514,13 +543,30 @@ flow (PaymentSandbox const& baseView,
 
             if (!best || best->quality < q ||
                 (best->quality == q && best->out < f.out))
-                best.emplace (f.in, f.out, std::move (*f.sandbox), *strand, q);
+            {
+                // If this strand is inactive (because it consumed too many
+                // offers) and ends up having the best quality, remove it from
+                // the activeStrands. If it doesn't end up having the best
+                // quality, keep it active.
+
+                if (f.inactive)
+                    markInactiveOnUse = activeStrands.size() - 1;
+                else
+                    markInactiveOnUse.reset();
+
+                best.emplace(f.in, f.out, std::move(*f.sandbox), *strand, q);
+            }
         }
 
         bool const shouldBreak = !bool(best);
```

Comment:
I know this is outside the scope, but we can just change `!bool(best)` to cast to `! static_cast<bool>(best)`

Reply:
Fixed. The cast isn't needed; I removed it. Also, I still tend to use c-style casts, which I know is a bad habit. I'll try to break it.